src/block/Lectern.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\tile\Lectern as TileLectern;
+use pocketmine\block\utils\BlockDataSerializer;
+use pocketmine\block\utils\FacesOppositePlacingPlayerTrait;
+use pocketmine\block\utils\HorizontalFacingTrait;
+use pocketmine\entity\Location;
+use pocketmine\entity\object\ItemEntity;
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+use pocketmine\item\WritableBookBase;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
+class Lectern extends Transparent {
+	use FacesOppositePlacingPlayerTrait;
+	use HorizontalFacingTrait;
+
+	protected bool $hasBook = false;
```

Comment:
This seems pointless, since the `book` field being null would represent this information anyway.

Reply:
I have $hasBook because of the lectern having it in its client blockState.

Reply:
It's not in the blockstate, it's in the tile.

Reply:
You are right I got them mixed up I removed it from the block